pyrobosim/pyrobosim/core/robot.py

```diff
@@ -946,15 +946,20 @@ def cancel_actions(self):
             self.logger.warning("There is no running action or plan to cancel.")
             return
 
-        if self.executing_nav and self.path_executor is not None:
-            self.logger.info("Canceling path execution...")
-            self.path_executor.cancel_execution = True
-            while self.executing_nav:
+        if self.executing_action:
+            if self.executing_nav and self.path_executor is not None:
```

Comment:
I think there are cases in which `self.executing_action` is false but `self.executing_nav` is true.

Basically, there are actions that can directly tell a robot to follow a path without going through the high-level action interface.

Reply:
You're right, I wasn't thinking of that. So is it better to add `self.canceling_execution = False` to the end of `Robot.execute_actions()`?

Since `Robot.cancel_actions()` is executed within a separate thread when using the ROS interface, it's difficult to make sure that `self.canceling_execution` is set to `True` before the execution callback resets the flag to `False`. If the cancellation thread does this after the execution callback performs `self.canceling_execution = False`, we enter an infinite loop.